core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,203 +16,246 @@
+          skillSummaries: response.skill_summaries,
+          topicUrlFragment: response.topic_url_fragment,
+          classroomUrlFragment: response.classroom_url_fragment
         });
+      }
+    }, errorResponse => {
+      if (errorCallback) {
+        errorCallback(errorResponse.error.error);
+      }
+    });
+  }
 
-      var putData = {
-        version: storyVersion,
-        commit_message: commitMessage,
-        change_dicts: changeList
-      };
-      $http.put(editableStoryDataUrl, putData).then(function(response) {
-        // The returned data is an updated story dict.
-        var story = angular.copy(response.data.story);
-
-        if (successCallback) {
-          successCallback(story);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _updateStory(
+      storyId: string, storyVersion: string,
+      commitMessage: string,
+      changeList: StoryChange[],
+      successCallback: (value: StoryBackendDict) => void,
+      errorCallback: (reason: string) => void): void {
+    const editableStoryDataUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
+        story_id: storyId
       });
+
+    const putData = {
+      version: storyVersion,
+      commit_message: commitMessage,
+      change_dicts: changeList
     };
 
-    var _changeStoryPublicationStatus = function(
-        storyId, newStoryStatusIsPublic, successCallback, errorCallback) {
-      var storyPublishUrl = UrlInterpolationService.interpolateUrl(
-        STORY_PUBLISH_URL_TEMPLATE, {
-          story_id: storyId
-        });
+    this.http.put<UpdateStoryBackendResponse>(
+      editableStoryDataUrl, putData).toPromise().then(
+      response => successCallback(response.story),
+      errorResponse => errorCallback(errorResponse.error.error)
+    );
+  }
 
-      var putData = {
-        new_story_status_is_public: newStoryStatusIsPublic
-      };
-      $http.put(storyPublishUrl, putData).then(function(response) {
-        if (successCallback) {
-          successCallback();
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _changeStoryPublicationStatus(
+      storyId: string,
+      newStoryStatusIsPublic: boolean,
+      successCallback: (value: number) =>void,
+      errorCallback: (reason: string) => void): void {
+    const storyPublishUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.STORY_PUBLISH_URL_TEMPLATE, {
+        story_id: storyId
       });
+    const putData = {
+      new_story_status_is_public: newStoryStatusIsPublic
     };
+    // eslint-disable-next-line max-len
+    this.http.put(storyPublishUrl, putData, { observe: 'response'}).toPromise().then(
+      response => successCallback(response.status),
```

Comment:
Earlier there was no `response.status`. Why is it now?

Reply:
corrected!

Reply:
Not corrected